### PR TITLE
Warn when a gabor file predates saved sigma (#219), and document the cumulative curve's endpoint (#220)

### LINKED ISSUE #219: Legacy test coverage for Gabor .Rdata files, incl. the missing-sigma fallback
`test-legacy-rdata.R` (added in #216) reads real 1.0.1 and 1.1.0 `.Rdata` files with the current code and pins the `nscales` compatibility fallback in `generateReferenceDistribution2IFC()`. Both fixtures use the default **sinusoidal** noise, which leaves two things untested.

## The missing-`sigma` fallback

`generateReferenceDistribution2IFC()` supplies a default `sigma` of 25 when the file lacks it (`R/generateReferenceDistribution.R`), exactly as it does for `nscales`. But `sigma` only enters the noise basis for **Gabor** noise — `generateNoisePattern()` passes it to `generateGabor()` and ignores it for sinusoids (`R/generateNoisePattern.R`). So no sinusoidal fixture can exercise that fallback, and changing the default from 25 to any other value leaves the current suite green while silently changing the reference distribution for legacy Gabor files.

Verified the gap is real: a Gabor stimulus set's reference norms move with `sigma` (25 → `1.01, 1.50, 1.27`; 10 → `0.65, 0.89, 0.72`), so a fixture could detect it. 1.0.1 generates Gabor files fine.

## Gabor legacy read-path in general

More broadly, `test-legacy-rdata.R` covers only the sinusoidal read path. A Gabor 1.0.1 file — which lacks both `nscales` and `sigma` — is the case that exercises both fallbacks at once, and is the natural fixture to add.

## Suggested shape

- Extend `tools/make-legacy-rdata.R` to emit a Gabor variant (a `(tag, noise_type, nscales)` table rather than the current `NSCALES` map), generated with the era's defaults so the fallbacks land on the correct basis.
- Add a test mirroring the `nscales` one: the fallback null equals an explicit `sigma = 25` and differs from an alternative.

Deferred from #216, where the scope was the sinusoidal path; the `NEWS.md` entry there was narrowed to `nscales` so it does not claim `sigma` coverage.

### LINKED ISSUE #220: [low confidence] computeCumulativeCICorrelation() does not aggregate repeated stimuli the way generateCI() does
## Confidence: LOW — likely intentional, filed for a deliberate check

This is an *observation from the repo-wide audit*, not a confirmed bug. My best guess is it's intentional, but it's worth confirming rather than assuming.

## Observation

When `participants = NA`, `generateCI()` collapses repeated presentations of the same stimulus before building the CI (`R/generateCI.R:184-191`):

```r
aggregated <- aggregate(responses, by=list(stimuli=stimuli), FUN=mean)
responses <- aggregated$x
stimuli   <- aggregated$stimuli
```

`computeCumulativeCICorrelation()` does no such aggregation — it uses the raw `stimuli`/`responses` in presentation order (`R/computeCumulativeCICorrelation.R:325, 334, 348`).

## Why this is probably fine

The function's whole purpose is a *cumulative* curve — correlation of the CI-so-far against the final CI as trials accumulate — so it has to walk trials in presentation order, and aggregating away that order would defeat it. So the difference is almost certainly by design.

## The one thing worth checking

When no `targetci` is supplied, the function computes its own `finalCI` from the un-aggregated data (`generateCINoise(params, responses, p)`, line 334). If a design ever presents the **same stimulus an unequal number of times**, that self-computed `finalCI` will not equal the CI that `generateCI()` would return for the same data (which aggregates first, then means). With equal repeat counts the two coincide; with unequal counts they diverge slightly. A user comparing the curve's endpoint against a `generateCI()` result could be mildly surprised.

Question to resolve: is the un-aggregated `finalCI` the intended reference, or should the default `finalCI` match `generateCI()`? If intended, a one-line note in `?computeCumulativeCICorrelation` would settle it; if not, aggregate only for the `finalCI` computation while keeping the cumulative loop on raw trial order.

_Static finding; R was not available to construct a divergent example and measure the difference._

### PR DESCRIPTION
**Pull request by rdotsch**
Closes #219, closes #220. The first change to go through `CONTRIBUTING.md` → "Plan first, in the
same pull request": the plan was this branch's first commit, reviewed over five rounds before any
implementation, and deleted here — so the squash leaves `main` one commit and no plan file, while
the plan and its four findings stay on this thread.

## #219 — the missing-`sigma` fallback was untestable, and silent where it mattered

`generateReferenceDistribution2IFC()` assumes `sigma = 25` for files predating 1.1.0, silently,
unlike the loud warnings for a missing `nscales` or `noise_type`. Both legacy fixtures use
sinusoidal noise, so **nothing in the suite could reach that fallback** — changing the default
from 25 left the suite green.

Measured, since `sigma` only reaches the basis through `generateGabor()`:

| noise | fallback | explicit 25 | explicit 10 |
|---|---|---|---|
| sinusoid | 0.7065, 0.6950, 0.7057 | identical | **identical** — inert |
| gabor | 0.6809, 0.6891, 0.6804 | identical | **0.6149, 0.6195, 0.6263** |

So the warning is conditioned on `noise_type`, not on the field alone: warning on sinusoidal files
would put a compatibility notice on the far more common legacy file about a parameter that
provably cannot change its result. `noise_type` is now resolved *before* `sigma`, since whether a
missing `sigma` matters at all depends on it.

`tools/make-legacy-rdata.R` grows a `(tag, noise_type, nscales, sigma)` table and emits a 1.0.1
gabor fixture — the case hitting both fallbacks at once, since 1.0.1 saves `noise_type` but
neither `nscales` nor `sigma` (verified in `git show v1.0.1:R/generateStimuli2IFC.R`, line 171).
`--versions` now also takes a row id, because a bare tag rewrites every fixture it covers, and
those are frozen artefacts whose embedded absolute paths differ per run.

The test pins **three** directions — warns when missing, silent when present, silent for
sinusoidal. The first alone is satisfied by an implementation that warns on every gabor file,
which would then warn spuriously on every current gabor stimulus set, fatal under
`options(warn = 2)`. Verified failing without the fix via `git stash push -- R/`: exactly one
failure, at the `expect_true` assertion.

## #220 — measured, then documented rather than changed

Filed low-confidence. The divergence is real and larger than "slight":

| design | identical | max abs pixel diff | correlation |
|---|---|---|---|
| equal repeats (2,2) | **yes**, bit-for-bit | 0 | 1.0 |
| unequal (3,1) | no | 0.065 | **0.845** |
| unequal (4,2,1,1) | no | 0.054 | **0.773** |

The user-facing part is sharper than the issue anticipated: with no `targetci` the function builds
its final CI from the same un-aggregated trials as the curve, so **the curve ends at exactly 1
whatever the data** — self-consistency that reads as convergence. On the same 4/2/1/1 data,
`targetci = generateCI(...)` ends it at 0.773.

Documented in `?computeCumulativeCICorrelation` and `DECISIONS.md`, and pinned by a test covering
both halves. Aggregating the self-computed final CI was rejected: it would move numeric output for
anyone calling without `targetci`, and stop the curve ending at 1 — a worse default than the one
being fixed.

The measurement raised a separate question, filed as #227 rather than settled here:
`generateCI()`'s aggregation weights each unique stimulus equally while the un-aggregated path
weights each trial equally, and its code comment frames that as a memory/processing optimisation,
which only holds where the two coincide.

## Verification

- Full suite **168 files, 535 assertions, 0 failures, 0 errors, 0 warnings, 0 skips**.
- Release gate `--quick`: **130 checks identical, 0 unexpected deviations**. The 2 expected ones
  pre-date this branch. No numeric output changes.
- `R CMD check --as-cran`: 2 NOTEs, both environmental (`New submission` with the known Medium
  403, and `unable to verify current time`). **`checking installed package size ... OK`** — the
  added 224 KB fixture does not trip it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by rdotsch** [comment]
@codex review

File: notes/plan-audit-batch-219-220.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+## Verification
+
+- `devtools::install()` first: tests calling `generateStimuli2IFC()` need the package installed,
+  not just `load_all()`-loaded, because its parallel workers `library(rcicr)` themselves.
+- `devtools::test()`, full suite green — `test-regression-baseline.R` and `test-legacy-rdata.R`
+  matter most here.
+- `Rscript tools/compare-release-output.R --quick`, to confirm the release gate is unaffected.
+- `R CMD build` in a temp directory, never at the repo root, then `R CMD check --as-cran`, watching
+  for an installed-size NOTE from the added fixture.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply notes/plan-audit-batch-219-220.md]: Good catch — the wording did contradict `CONTRIBUTING.md` → Releasing, and it is fixed in bef34e4.

Not by switching to `R CMD build .` at the root, though, because both rules are real and they are about different things. The release tarball is built with `R CMD build .` at the repo root and never in a worktree, which is what `CONTRIBUTING.md:328` and the `^\.git$` entry are about. Separately, `R CMD build .` at the root **overwrites an externally-checked tarball sitting there**, which has happened here — that is the rule the plan was reaching for and stated too broadly.

The form that satisfies both is neither a copy nor a worktree:

```sh
tmp=$(mktemp -d) && (cd "$tmp" && R CMD build /workspaces/rcicr)
```

`R CMD build [options] pkgdirs` takes the package directory as an argument and writes the tarball into the *working* directory. So the source tree is the real repo root — `.git` is a directory and `.Rbuildignore` behaves exactly as it does for a release — while the tarball lands outside the root. The plan now spells that out and names the release procedure as unchanged.

**Comment by rdotsch** [comment]
@codex review

File: notes/plan-audit-batch-219-220.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+- `devtools::install()` first: tests calling `generateStimuli2IFC()` need the package installed,
+  not just `load_all()`-loaded, because its parallel workers `library(rcicr)` themselves.
+- `devtools::test()`, full suite green — `test-regression-baseline.R` and `test-legacy-rdata.R`
+  matter most here.
+- `Rscript tools/compare-release-output.R --quick`, to confirm the release gate is unaffected.
+- A tarball build, then `R CMD check --as-cran` on it, watching for an installed-size NOTE from
+  the added fixture. This is an ad-hoc build rather than a release one, so run it as:
+
+  ```sh
+  tmp=$(mktemp -d) && (cd "$tmp" && R CMD build /workspaces/rcicr)
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply notes/plan-audit-batch-219-220.md]: Right, and it is the repo's own rule — `AGENTS.md` asks for claims that survive on someone else's machine, and an absolute path from the box the plan was written on is exactly that failure. Fixed in 6a849d4:

```sh
pkg=$(pwd)                                        # at the repo root
tmp=$(mktemp -d) && (cd "$tmp" && R CMD build "$pkg")
```

Capturing the root before changing directory keeps the property the step needs — the source tree is the real checkout, not a copy or a worktree — without naming where that checkout lives.

**Comment by rdotsch** [comment]
@codex review

File: notes/plan-audit-batch-219-220.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+`generateCINoise()` takes `colMeans(stimuli * responses)`
+([`generateCINoise.R:20-28`](../R/generateCINoise.R#L20-L28)), so with equal repeat counts the two
+are algebraically identical and with unequal counts they diverge. Confirm that empirically and
+quantify it — correlation and max absolute pixel difference between the two CIs, and the effect on
+the curve's endpoint, which is what a user would actually notice. Check the equal-count case
+agrees exactly, so the result distinguishes "only under unequal counts" from "always".
+
+**Then stop and report the numbers**, before any documentation, [`DECISIONS.md`](../DECISIONS.md)
+entry or test is written. A fix that changed numeric output is ruled out in advance by the guiding
+constraint.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply notes/plan-audit-batch-219-220.md]: Agreed, and this one was my reasoning rather than a typo. Fixed in c915ff8.

`AGENTS.md:136` reads "Never change existing call syntax, argument meanings, or the numeric output of a function **silently**", and then names the path for when it does change: a `NEWS.md` "Reproducibility impact" entry, with the release gate agreeing. I had collapsed that into a blanket prohibition, which could have left a confirmed defect standing on a technicality.

The plan now names both outcomes — a doc note plus a `DECISIONS.md` entry if the difference is what the issue expects, or a correction if the numbers show the self-computed `finalCI` is wrong — and says explicitly that a correction is not ruled out in advance. The checkpoint is unchanged: what it forbids is deciding either way before the numbers exist.

**Comment by rdotsch** [comment]
@codex review

File: notes/plan-audit-batch-219-220.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+and `legacy_fixture()` are keyed on version strings and need a variant-aware form.
+
+- The Gabor 1.0.1 fixture lacks `nscales` and `sigma` and carries `noise_type == "gabor"`.
+- The current `generateCI()` reads it and returns a real CI.
+- **The missing-`sigma` fallback picks 25 specifically**: the fallback null is `identical()` to an
+  explicit `sigma = 25` and differs from an explicit `sigma = 10`. The *differs* half is what
+  stops the test passing vacuously — finite norms would not, per CONTRIBUTING's "a test's title is
+  a claim".
+- The new warning fires for the Gabor file and does **not** fire for the sinusoidal 1.0.1 fixture.
+  Match a fragment of the real `conditionMessage()`.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply notes/plan-audit-batch-219-220.md]: Real hole, and the sharpest of the four. Fixed in da4d852.

The two assertions were "fires for the Gabor fixture, silent for the sinusoidal one", and an implementation that warned on *every* Gabor file passes both — at which point every current Gabor stimulus set, all of which save `sigma`, collects a spurious compatibility warning, fatal under `options(warn = 2)`. That is a live regression the spec would have waved through, not a theoretical one.

The separating case is a Gabor file that does carry `sigma`, and the fallback test already builds one as its `explicit_25` copy, so the fix is a single extra assertion on an object that exists. The plan now names all three directions: warns when missing, silent when present, silent for sinusoidal regardless.

This is `CONTRIBUTING.md` → "a test's title is a claim; make the assertions actually support it", applied to a case I had missed.

**Comment by rdotsch** [comment]
@codex review

**Comment by rdotsch** [comment]
@codex review

File: R/computeCumulativeCICorrelation.R [hunk 5]
```diff
@@ -4,6 +4,23 @@
...
 #'
+#' @section Repeated presentations of the same stimulus:
+#' This function walks trials in the order they were presented and does not aggregate repeated
+#' presentations of a stimulus, unlike \code{\link{generateCI}}, which averages the responses to
+#' each unique stimulus before building its classification image. That is deliberate: collapsing
+#' repeats would discard the presentation order a cumulative curve is entirely about.
+#'
+#' One consequence is worth knowing. With no \code{targetci}, the final CI computed here is built
+#' from the same un-aggregated trials as the curve, so the curve ends at exactly 1 by
+#' construction -- that is self-consistency, not evidence of convergence. Where every stimulus was
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply R/computeCumulativeCICorrelation.R]: Correct, and measured before fixing: with six responses the curve ends at **0.967** at `step = 2` and **0.938** at `step = 3`, because `seq(1, length(responses), step)` evaluates trials 1/3/5 and 1/4 respectively, so the last response never enters a cumulative CI even though it is in the final CI being compared against. Fixed in 7e11cd7.

Documented rather than changed. Appending the final trial to the sequence would alter the returned vector for every `step > 1` caller, which `AGENTS.md` puts behind a `NEWS.md` "Reproducibility impact" entry and the release gate — a steep price for a curve read for its shape, and it would also make the returned length no longer follow from `step` alone.

The unqualified claim was in three places — the roxygen, `NEWS.md` and `DECISIONS.md` — and all three now carry the qualification. A test pins both halves: `step = 1` gives six points ending at 1, `step = 2` gives three and ends below it.

**Comment by rdotsch** [comment]
@codex review

File: R/computeCumulativeCICorrelation.R [hunk 6]
```diff
@@ -4,6 +4,29 @@
...
+#' @section Repeated presentations of the same stimulus:
+#' This function walks trials in the order they were presented and does not aggregate repeated
+#' presentations of a stimulus, unlike \code{\link{generateCI}}, which averages the responses to
+#' each unique stimulus before building its classification image. That is deliberate: collapsing
+#' repeats would discard the presentation order a cumulative curve is entirely about.
+#'
+#' One consequence is worth knowing. With no \code{targetci}, the final CI computed here is built
+#' from the same un-aggregated trials as the curve. Where the evaluated trials reach the last one
+#' -- always so at the default \code{step = 1} -- the curve's final point compares that CI with
+#' itself and is exactly 1: self-consistency, not evidence of convergence. A larger \code{step}
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply R/computeCumulativeCICorrelation.R:16]: Confirmed, and it is broader than the endpoint. Measured on a six-trial set with each stimulus answered both ways: the final CI is uniformly zero, and correlating against a constant is undefined, so **every** point on the curve is `NA` — not just the last one failing to be 1. Fixed in 5c23fe7.

The roxygen, `NEWS.md` and `DECISIONS.md` now qualify the guarantee on the CI varying at all, and say plainly that an all-`NA` curve means the responses carry no net signal rather than that the call failed — which is the part a user hitting this would actually need.

The test asserts `all(final_ci == 0)` before asserting the `NA`s, so a future change that made the CI non-degenerate cannot leave it passing for the wrong reason.

Full suite is 170 files, 542 assertions, 0 failures.

**Comment by rdotsch** [comment]
@codex review

File: R/computeCumulativeCICorrelation.R [hunk 7]
```diff
@@ -4,6 +4,35 @@
...
+#' can stop short, because trials are taken at \code{seq(1, length(responses), step)}: with six
+#' responses and \code{step = 2} the last one evaluated is the fifth, and the curve ends at
+#' whatever that partial CI correlates to -- 0.97 in one such set, not 1.
+#'
+#' Both statements assume the CI being compared against varies at all. Responses that cancel
+#' exactly -- every presentation of a stimulus answered both ways -- average to a uniformly zero
+#' CI, and a correlation against a constant is undefined, so \strong{every} point on the curve is
+#' \code{NA} rather than the last one being 1. An all-\code{NA} curve means the responses carry no
+#' net signal, not that the call failed.
```
[the remark this answers is not in the document]
Comment (rdotsch) [reply R/computeCumulativeCICorrelation.R]: Confirmed, and this one is a real bug rather than a wording problem — thank you. Measured: masking a **4x4 corner of a 32x32 CI, 16 of 1024 pixels, empties the entire curve** while the same data with an unmasked target gives real correlations throughout. My new recommendation would have walked users straight into it.

Documented in bba7094 and filed as #228. The recommendation now says to build the target without a mask, and the all-`NA` diagnosis distinguishes its two causes: cancelled-out responses, which carry no net signal, and a masked target, which carries plenty.

Not fixed here, deliberately. `use = "complete.obs"` would correlate over the unmasked pixels and leave every unmasked call bit-identical — but it changes returned values from `NA` to real numbers, and whether the mask should apply to one side or to both is worth settling before picking. Under `CONTRIBUTING.md` → "Plan first, in the same pull request" that is a behaviour change deserving its own plan, and this PR is already two issues wide.

The test pins the current behaviour as documented and asserts the unmasked target gives a real curve on the same data, so the `NA`s are demonstrated to come from the mask rather than the responses. #228 notes that fixing it means updating that test, which is the intended signal.

**Comment by rdotsch** [comment]
@codex review

**Comment by rdotsch** [comment]
@codex review
